### PR TITLE
Frontend ci

### PR DESCRIPTION
**Pull request by peterjah**

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -1,7 +1,5 @@
 on:
   push:
-    branches: [ main ]
-  pull_request:
```
Comment (Thykof) [inline .github/workflows/tests.yml:4]: do you want to keep that or is it just for you CI tests?

**Review by Thykof — APPROVED** [review]

**Review by SebastienValla — COMMENTED** [review]
LGTM !

File: gen-frontend.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package resource
```
Comment (gregLibert) [inline gen-frontend.go:1]: why not put all that in gen.go at the root of the project?

File: .github/actions/install/action.yml [hunk 3]
```diff
@@ -14,6 +13,18 @@ runs:
...
+    - name: Install Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: 18
+        cache: 'npm'
+        cache-dependency-path: web/thyra-frontend/package-lock.json
+
+    - name: Build react frontend
+      shell: bash
+      run: go generate gen-frontend.go
```
Comment (gregLibert) [inline .github/actions/install/action.yml:30]: why not regenerating everything `go generate ./...` ?

**Review by gregLibert — COMMENTED** [review]
:question:

File: .github/workflows/tests.yml (hunk 1 above)
Comment (peterjah) [reply .github/workflows/tests.yml:4]: yes i want to keep! 😄 
the previous behaviour is to execute test & lint only at PR creation or on push on main.
But i think its better to execute it at every push on every branch

File: gen-frontend.go (hunk 2 above)
Comment (peterjah) [reply gen-frontend.go:1]: issue created:
https://github.com/massalabs/thyra/issues/461

**Pull request merged**